.gitlab/darwin/toolchain.nix

```diff
@@ -0,0 +1,34 @@
+{ system }:
```

Comment:
what is this for? what does it do? etc. etc. documentation please!

Reply:
This is taken almost verbatim from the GHC nix setup script. I can add a note pointing to that, but beyond that I have very little experience with nix, so all of the nix stuff is just cargo-culted from GHC itself.

Reply:
"I cargo-culted this from GHC, here's a link to the upstream version" is _incredibly_ useful information for someone reading this!